docker/Dockerfile

```diff
@@ -25,20 +26,13 @@ RUN crystal spec --warnings all \
     --link-flags "-lxml2 -llzma"
 
 RUN if [[ "${release}" == 1 && "${disable_quic}" == 1 ]] ; then \
-        crystal build ./src/invidious.cr \
-        --release \
-        -Ddisable_quic \
-        --static --warnings all \
-        --link-flags "-lxml2 -llzma"; \
+        make DISABLE_QUIC=1 \
     elif [[ "${release}" == 1 ]] ; then \
-        crystal build ./src/invidious.cr \
-        --release \
-        --static --warnings all \
-        --link-flags "-lxml2 -llzma"; \
+        make RELEASE=1 \
+    elif [[ "${api_only}" == 1 ]] ; then \
+        make RELEASE=1 API_ONLY=1  \
     else \
-        crystal build ./src/invidious.cr \
-        --static --warnings all \
-        --link-flags "-lxml2 -llzma"; \
+        make RELEASE=0 \
```

Comment:
you could simplify all of this to a single line:
`RUN make "RELEASE=${release}" "DISABLE_QUIC=${disable_quic}" "API_ONLY=${api_only}"`